packages/postgres/src/error.ts

```diff
@@ -0,0 +1,22 @@
+export function formatQueryWithErrorPosition(query: string, position: number | null | undefined) {
+  if (position == null) return query;
+
+  const preSql = query.substring(0, position);
+  const postSql = query.substring(position);
+  const prevNewline = Math.max(0, preSql.lastIndexOf('\n') + 1);
+  let nextNewline = postSql.indexOf('\n');
+  if (nextNewline < 0) nextNewline = postSql.length;
+  nextNewline += preSql.length;
```

Comment:
Nit: `preSql` and `postSql` are not needed at all in this computation, if you use the second argument of `indexOf` and `lastIndexOf`.

```suggestion
  const prevNewline = Math.max(0, query.lastIndexOf('\n', position) + 1);
  let nextNewline = query.indexOf('\n', position);
  if (nextNewline < 0) nextNewline = query.length;
```